core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -95,11 +101,18 @@ public OverwriteFiles caseSensitive(boolean isCaseSensitive) {
   @Override
   public OverwriteFiles validateNoConflictingAppends(Expression newConflictDetectionFilter) {
     Preconditions.checkArgument(newConflictDetectionFilter != null, "Conflict detection filter cannot be null");
-    this.conflictDetectionFilter = newConflictDetectionFilter;
+    this.appendConflictDetectionFilter = newConflictDetectionFilter;
     failMissingDeletePaths();
```

Comment:
Question: Does this call to `failMissingDeletePaths()` still belong here or should it be moved to the `validateNoConflictingDeleteFiles` call?

Reply:
I'll double check.